src/phsoar_null_router/soar_null_router_connector.py

```diff
@@ -188,10 +188,15 @@ def _handle_test_connectivity(self, param):
     
     def _handle_block(self, param):
         action_result = self.add_action_result(ActionResult(dict(param)))
-        cidr = phantom.get_req_value(param, "cidr")
-        result = self._bhr.block(cidr=cidr, source='SOAR', why='Appears in our suspicious event list.')
 
-        return action_result.set_status(phantom.APP_SUCCESS, f"Blocked {cidr}")
+        args = {}
+        for key in ["cidr", "source", "why", "duration"]:
+            args[key] = phantom.get_req_value(param, key)
+        
+        result = self._bhr.block(**args)
```

Comment:
I think this can more simply be:

```suggestion
        result = self._bhr.block(** param)
```